metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py

```diff
@@ -189,11 +193,14 @@ def get_fields(self, delta_table: DeltaTable) -> List[SchemaField]:
         for raw_field in delta_table.schema().fields:
             # parse the complex fields like array, structure and get the native type
             # feed that native type and get the avro schema
-            if raw_field.type.type in ["array", "struct"]:
+            if raw_field.type.type == "array":
+                complex_fields = complex_fields + self._parse_datatype(
+                    raw_field.to_json()
+                )
+            elif raw_field.type.type == "struct":
```

Comment:
This is not what I intended to change. I was mentioning about adding such conditions within `_inner_field_details` method instead of (in addition to) relying on less-explanatory ` if "elementType" in raw_field and type(raw_field.get("elementType")) == dict:` Or whether ` elif "fields" in raw_field and type(raw_field.get("fields")) == list:` .

`

Reply:
reason of having the conditions within `get_fields` is, for `non-compled primitive` types we are directly creating the `SchemaField` and for only `complex` fields we are first creating the correct `native datatype` string and then creating `avro SchemaField`